src/Test/Perf/ConsumptionParser/ConsumptionParser.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace Roslyn.Test.Performance.Utilities.ConsumptionParser
+{
+
+    internal struct ConsumptionParse
+    {
+        public IEnumerable<ScenarioResult> Scenarios { get; }
+        public ConsumptionParse(IEnumerable<ScenarioResult> scenarios)
+        {
+            this.Scenarios = scenarios;
+        }
+
+        public static ConsumptionParse Parse(String s)
+        {
+            var document = XDocument.Parse(s);
+
+            var scenariosOut = new List<ScenarioResult>();
+            foreach (var scenario in document.Descendants("ScenarioResult"))
+            {
+                var countersOut = new List<CounterResult>();
+                var name = scenario.Attribute("Name").Value;
+                if (name == "..TestDiagnostics..")
+                {
+                    continue;
+                }
+
+                foreach (var counter in scenario.Descendants("CounterResult"))
+                {
+                    countersOut.Add(new CounterResult(
+                        name: counter.Attribute("Name").Value,
+                        units: counter.Attribute("Units").Value,
+                        isDefault: bool.Parse(counter.Attribute("Default").Value),
+                        top: bool.Parse(counter.Attribute("Top").Value),
+                        iteration: int.Parse(counter.Attribute("Iteration").Value),
+                        value: counter.Value
+                    ));
+                }
+                // Sort the metric by name
+                countersOut.Sort((a, b) => a.Name.CompareTo(b.Name));
+                scenariosOut.Add(new ScenarioResult(name, countersOut));
+            }
+
+            // Sort the scenarios by name
+            scenariosOut.Sort((a, b) => a.Name.CompareTo(b.Name));
+            return new ConsumptionParse(scenariosOut);
+        }
+
+    }
+
+    internal struct ScenarioResult
+    {
+        public ScenarioResult(string name, IEnumerable<CounterResult> counters)
+        {
+            this.Name = name;
+            this.Counters = counters;
+        }
+
+        public CounterResult? this[string name]
+        {
+            get
+            {
+                foreach (var counter in Counters)
```

Comment:
Did we consider storing these values as a `SmallDictionary` or hash table of some sort?  If there aren't a lot of these values this should be fine.  Want to make sure we considered the perf implications of linear lookup here.